pkg/controller/clusterstate/cache/cache.go

```diff
@@ -261,13 +261,15 @@ func (sc *ClusterStateCache) updateState() error {
 	idleMin := api.EmptyResource()
 	idleMax := api.EmptyResource()
 
-	firstNode :=  true
+	firstNode := true
 	for _, value := range cluster.Nodes {
 		// Do not use any Unschedulable nodes in calculations
-		if value.Unschedulable == true {
-			klog.V(6).Infof("[updateState] %s is marked as unschedulable node Total: %v, Used: %v, and Idle: %v will not be included in cluster state calculation.",
-				value.Name, value.Allocatable, value.Used, value.Idle)
-			continue
+		if !value.Unschedulable {
+			if value.IsReady == "true" {
```

Comment:
```
if value.Unschedulable == true || value.IsReady != v1.ConditionTrue {
```

Suggest change to handle nodes "bad" nodes scenario below:
Unschedulable = true  IsReady = true
Unschedulable = false  IsReady = false
Suggest change to let "good" nodes continue past this check for the scenario below:
Unschedulable = false  IsReady = true

Reply:
agree, thanks now good nodes do not get skipped. and problem nodes are skipped